### ISSUE TITLE
Modify 'not-space' rule to exclude '}'

### ISSUE THREAD

**Issue by codalogic** [arineng/jcrvalidator#101]
We have an ABNF rule of:

    ruleset-id       = ALPHA *not-space

But when ruleset-id in a multi-line-directive form, such as:

    #{ruleset-id foo}

the closing `}` will be interpreted as part of the `ruleset-id`.

Is that acceptable, or should we try to protect users from such issues?

**Comment by anewton1998** [comment]
Hmmm.... I thought we had parser tests for that but apparently not. I think we should modify the ABNF so avoid consuming the `}`.

**Comment by codalogic** [comment]
I've looked a bit further.

I've double checked that `}` can't appear in URIs, so we're good there.

I briefly wondered whether to make it escapable (eg `#{ruleset-id abc\}def}`), but since it can't appear in URIs, it doesn't seem worth the effort.

I did wonder whether we should also treat `;` at the beginning of comments specially, but these are permitted in URIs, and it's usual to include space before a comment anyway. So, decided against this.

It looks easy to do.

So I'll have a go at it.
